activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -332,10 +332,21 @@ def load_schema(configuration, format = ActiveRecord::Base.schema_format, file =
         end
         ActiveRecord::InternalMetadata.create_table
         ActiveRecord::InternalMetadata[:environment] = environment
+        ActiveRecord::InternalMetadata[:schema_sha1] = schema_sha1(file)
       ensure
         Migration.verbose = verbose_was
       end
 
+      def schema_up_to_date?(configuration, format = ActiveRecord::Base.schema_format, file = nil, environment = env, spec_name = "primary")
+        file ||= dump_filename(spec_name, format)
+
+        return true unless File.exist?(file)
+
+        ActiveRecord::Base.establish_connection(configuration)
+        return false unless ActiveRecord::InternalMetadata.table_exists?
+        ActiveRecord::InternalMetadata[:schema_sha1] == schema_sha1(file)
```

Comment:
Do we need to stop this information in the database? Can't it be a file on tmp?

Reply:
Good question ❤️ 

I think the database is the best place for it, we want to store this exactly per-database: `my_app_test` separate from `my_app_development`, `my_app_test-1` separate from `my_app_test-2` when using parallelism. So storing it _in_ the database itself I think accomplishes this the best. It also handles databases being dropped/restored from backup/etc.

I also think it's similar enough to the other information we store in the DB: very similar to schema migrations (metadata about the "structure" of the DB) and somewhat similar to `environment` (it records the state at creation/load) that it "feels" right to me.